test/tests/queries/variables.test.js

```diff
@@ -280,5 +280,247 @@ describe('graphql - variables', () => {
+              nodes {
+                ID
+                title
+              }
+            }
+          }
+        }
+      `
+      const variables = { filter: null }
+      const data = {
+        AdminService: {
+          Books: {
+            nodes: [
+              { ID: 201, title: 'Wuthering Heights' },
+              { ID: 207, title: 'Jane Eyre' },
+              { ID: 251, title: 'The Raven' },
+              { ID: 252, title: 'Eleonora' },
+              { ID: 271, title: 'Catweazle' }
+            ]
+          }
+        }
+      }
+      const response = await POST('/graphql', { query, variables })
+      expect(response.data).toEqual({ data })
+    })
+
+    test('query variable with undefined value where an object value is expected', async () => {
+      const query = gql`
+        query ($filter: [AdminService_Books_filter]) {
+          AdminService {
+            Books(filter: $filter) {
+              nodes {
+                ID
+                title
+              }
+            }
+          }
+        }
+      `
+      const variables = { filter: { stock: undefined } }
+      const data = {
+        AdminService: {
+          Books: {
+            nodes: [
+              { ID: 201, title: 'Wuthering Heights' },
+              { ID: 207, title: 'Jane Eyre' },
+              { ID: 251, title: 'The Raven' },
+              { ID: 252, title: 'Eleonora' },
+              { ID: 271, title: 'Catweazle' }
+            ]
+          }
+        }
+      }
+      const response = await POST('/graphql', { query, variables })
+      expect(response.data).toEqual({ data })
+    })
+
+    test('query variable with null value where an object value is expected', async () => {
+      const query = gql`
+        query ($filter: [AdminService_Books_filter]) {
+          AdminService {
+            Books(filter: $filter) {
+              nodes {
+                ID
+                title
+              }
+            }
+          }
+        }
+      `
+      const variables = { filter: { stock: null } }
+      const data = {
+        AdminService: {
+          Books: {
+            nodes: [
+              { ID: 201, title: 'Wuthering Heights' },
+              { ID: 207, title: 'Jane Eyre' },
+              { ID: 251, title: 'The Raven' },
+              { ID: 252, title: 'Eleonora' },
+              { ID: 271, title: 'Catweazle' }
+            ]
+          }
+        }
+      }
+      const response = await POST('/graphql', { query, variables })
+      expect(response.data).toEqual({ data })
+    })
+
+    test('query variable with undefined value as a field of an argument', async () => {
+      const query = gql`
+        query ($filter: [AdminService_Books_filter]) {
+          AdminService {
+            Books(filter: $filter) {
+              nodes {
+                ID
+                title
+              }
+            }
+          }
+        }
+      `
+      const variables = { filter: { stock: { eq: undefined } } }
```

Comment:
is that really something we want to support? Looks super weird to me.

Reply:
I don't think we really have a choice since `undefined` can always be used as an JavaScript object value, but gets JSON stringified to `{"filter":{"stock":{}}}`, which means it evaluates to `true`. Or did you mean something else?

Reply:
> gets JSON stringified to {"filter":{"stock":{}}}

I see, then the same rules apply. Okay, then it's fine.